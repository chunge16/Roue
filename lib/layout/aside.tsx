import * as React from 'react';

const aside:React.FunctionComponent= ()=>{
    return (
        <div>
           aside
        </div>
    )
};
export default  aside