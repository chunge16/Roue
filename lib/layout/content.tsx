import * as React from "react";
import {scopedClassMaker} from "../helper/classes";
const sc = scopedClassMaker("roue-layout");

const content:React.FunctionComponent= ()=>{
    return (
        <div className={sc("content")}>
            content
        </div>
    )
};
export default  content