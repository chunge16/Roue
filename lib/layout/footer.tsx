import * as React from 'react'

const footer:React.FunctionComponent= ()=>{
    return (
        <div>
            footer
        </div>
    )
};
export default  footer