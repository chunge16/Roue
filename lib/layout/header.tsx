import * as React from "react";
import {scopedClassMaker} from "../helper/classes";
const sc = scopedClassMaker('roue-layout');

const header:React.FunctionComponent= (props)=>{
    return (
        <div className={sc("header")}>
            {props.children}
        </div>
    )
};
export default  header