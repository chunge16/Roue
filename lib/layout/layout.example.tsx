import * as React from 'react';
import Layout from './layout';
import Header from './header';
import Content from './content';
import Footer from './footer';
import Aside from "./aside";

export default function () {

    return (
        <div>
            <div>
                <h1>example 1 </h1>
                <Layout style={{height: 500}}>
                    <Header>header</Header>
                    <Content>content</Content>
                    <Footer>footer</Footer>
                </Layout>
            </div>
            <div>
                <h1>example 2 </h1>
                <Layout style={{height: 500}}>
                    <Header>header</Header>
                    <Layout>
                        <Aside>aside</Aside>
                        <Content>content</Content>
                    </Layout>
                    <Footer>footer</Footer>
                </Layout>
            </div>
            <div>
                <h1>example 3 </h1>
                <Layout style={{height: 500}}>
                    <Header>header</Header>
                    <Layout>
                        <Content>content</Content>
                        <Aside>aside</Aside>
                    </Layout>
                    <Footer>footer</Footer>
                </Layout>
            </div>
            <div>
                <h1>example 4 </h1>
                <Layout style={{height: 500}}>
                    <Aside>aside</Aside>
                    <Layout>
                        <Header>header</Header>
                        <Content>content</Content>
                        <Footer>footer</Footer>
                    </Layout>
                </Layout>
            </div>
            <div>
                <h1>example 5 </h1>
                <Layout style={{height: 500}}>
                    <Layout>
                        <Header>header</Header>
                        <Content>content</Content>
                        <Footer>footer</Footer>
                    </Layout>
                    <Aside>aside</Aside>
                </Layout>
            </div>
        </div>
    )
}