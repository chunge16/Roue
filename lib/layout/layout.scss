.roue-layout {
  border: 1px solid red;
  display: flex;
  flex-direction: column;

  &-content {
    border: 1px solid red ;
    flex-grow: 1;
  }
  & &{
    flex-grow: 1;
  }
}
.hasAside{
  flex-direction: row;
}