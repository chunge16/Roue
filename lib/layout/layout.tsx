import * as React from 'react'
import {ReactElement} from "react";
import './layout.scss'

interface Props {
    children:ReactElement|Array<ReactElement>
}

const x:React.FunctionComponent<Props>= (props)=>{
    return (
        <div>
            {props.children}
        </div>
    )
};
export default  x