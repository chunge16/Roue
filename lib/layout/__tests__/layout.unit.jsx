import * as renderer from 'react-test-renderer'
import * as React from 'react'
import {mount} from 'enzyme'
import {Layout, Header, Content, Aside, Footer} from "../layout";


describe("Layout", () => {
    it("layout exist ", () => {
        const json = renderer.create(
            <Layout>
                <Header>header</Header>
                <Content>content</Content>
                <Footer>footer</Footer>
            </Layout>)
        expect(json).toMatchSnapshot();
    })
    it("layout hasAside ", () => {
        const component = mount(
            <Layout>
                <Header>header</Header>
                <Layout>
                    <Aside>aside</Aside>
                    <Content>content</Content>
                </Layout>
                <Footer>footer</Footer>
            </Layout>);
        expect(component.exists(".roue-layout-hasAside")).toEqual(true)
    })
})