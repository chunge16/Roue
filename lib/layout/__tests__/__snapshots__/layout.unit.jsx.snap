// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout layout exist  1`] = `
<div
  className="roue-layout"
>
  <div
    className="roue-layout-header"
  >
    header
  </div>
  <div
    className="roue-layout-content"
  >
    content
  </div>
  <div
    className="roue-layout-footer"
  >
    footer
  </div>
</div>
`;
